src/assets/scripts/client/aircraft/StripView/StripViewController.js

```diff
@@ -129,12 +129,30 @@ export default class StripViewController {
             const aircraftModel = aircraftList[i];
             const stripViewModel = this._collection.findStripByAircraftId(aircraftModel.id);
 
+            // we end up here when an aircraft is outside controlled airspace
+            // but not yet removed from the view
+            if (typeof stripViewModel === 'undefined') {
+                return;
+            }
+
             if (aircraftModel.inside_ctr && !stripViewModel.insideCenter) {
                 this._addViewToStripList(stripViewModel);
+
+                return;
+            }
+
+            // an aircraft is exiting controlled airspace but the aircraft strip
+            // is still in the DOM. this where the strip gets removed from the DOM
+            if (!aircraftModel.inside_ctr && aircraftModel.isRemovable) {
+                this.removeStripView(aircraftModel);
```

Comment:
Wouldn't it make more sense to remove the strip upon `AircraftModel.onAirspaceExit()` and `AircraftController.aircraft_remove()` instead of using conditional logic within the StripViewModel itself?

Reply:
The `AircraftModel` has no knowledge of the `StripView`, only the `AircraftController` does and `.aircraft_remove()` only fires once an aircraft lands or is way outside controlled airspace.

The removing logic should be consolidated and updated first before this PR does its thing.

Reply:
We could also perhaps use event bus callbacks? `EVENT.EXIT_AIRSPACE` and `EVENT.REMOVE_AIRCRAFT` or similar, which call the appropriate methods? Just spitballing anyway.

Reply:
We could do that, but we'd still have some spaghetti code.  see: #916 for future work on all that